src/status_im/bots/events.cljs

```diff
@@ -0,0 +1,99 @@
+(ns status-im.bots.events
+  (:require [re-frame.core :refer [trim-v]]
+            [status-im.utils.handlers :refer [register-handler-db register-handler-fx]]))
+
+;;;; Helper fns
+
+(defn- chats-with-bot [chats bot]
+  (reduce (fn [acc [_ {:keys [chat-id contacts]}]]
+            (let [contacts (map :identity contacts)]
+              (if (some #{bot} contacts)
+                (conj acc chat-id)
+                acc)))
+          []
+          chats))
+
+;; TODO: optime this, for sure we don't need to re-calculate all bot subscriptions every time something in bot db changes
+(defn- check-subscriptions-fx
+  [db {:keys [bot key path value]}]
+  (let [{:keys [bot-db chats]} db
+        path'                  (or path [key])
+        subscriptions          (get-in db [:bot-subscriptions path'])
+        current-bot-db         (get bot-db bot)]
+    {:call-jail-function-n
+     (for [{:keys [bot subscriptions name]} subscriptions
+           :let [subs-values (reduce
+                              (fn [acc [sub-name sub-path]]
```

Comment:
Can we split this big function into smaller ones? Would help with readability, documentation and testability.

Reply:
Sure, will do